app/code/Magento/InventorySales/Model/IsProductSalableForRequestedQtyCondition/IsCorrectQtyCondition.php

```diff
@@ -0,0 +1,104 @@
+
+/**
+ * @inheritdoc
+ */
+class IsCorrectQtyCondition implements IsProductSalableForRequestedQtyInterface
+{
+    /**
+     * @var GetStockItemConfigurationInterface
+     */
+    private $getStockItemConfiguration;
+
+    /**
+     * @var GetReservationsQuantityInterface
+     */
+    private $getReservationsQuantity;
+
+    /**
+     * @var GetStockItemDataInterface
+     */
+    private $getStockItemData;
+
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    /**
+     * @var MathDivision
+     */
+    protected $mathDivision;
+
+    public function __construct(
+        GetStockItemConfigurationInterface $getStockItemConfiguration,
+        Configuration $configuration,
+        GetReservationsQuantityInterface $getReservationsQuantity,
+        GetStockItemDataInterface $getStockItemData,
+        MathDivision $mathDivision
+    ) {
+        $this->getStockItemConfiguration = $getStockItemConfiguration;
+        $this->configuration = $configuration;
+        $this->getStockItemData = $getStockItemData;
+        $this->getReservationsQuantity = $getReservationsQuantity;
+        $this->mathDivision = $mathDivision;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute(string $sku, int $stockId, float $requestedQty): bool
+    {
+        $stockItemConfiguration = $this->getStockItemConfiguration->execute($sku, $stockId);
+        if (null === $stockItemConfiguration) {
+            return false;
+        }
+
+        $stockItemData = $this->getStockItemData->execute($sku, $stockId);
+
+        $qtyWithReservation = $stockItemData['quantity'] + $this->getReservationsQuantity->execute($sku, $stockId);
+        if ($requestedQty > $qtyWithReservation) {
+            return false;
+        }
+
+        $globalMinQty = $this->configuration->getMinQty();
+        if (($stockItemConfiguration->isUseConfigMinQty() == 1 && $requestedQty < $globalMinQty)
+            || ($stockItemConfiguration->isUseConfigMinQty() == 0
+                && $requestedQty < $stockItemConfiguration->getMinQty()
+            )) {
+            return false;
+        }
+
+        $globalMaxSaleQty = $this->configuration->getMaxSaleQty();
+        if (($stockItemConfiguration->isUseConfigMaxSaleQty() == 1 && $requestedQty > $globalMaxSaleQty)
+            || ($stockItemConfiguration->isUseConfigMaxSaleQty() == 0
+                && $requestedQty > $stockItemConfiguration->getMaxSaleQty()
+            )) {
+            return false;
+        }
+
+        $globalEnableQtyIncrements = $this->configuration->getEnableQtyIncrements();
+        if (($globalEnableQtyIncrements
+                && $this->mathDivision->getExactDivision($requestedQty, $this->configuration->getQtyIncrements()))
+            || ($stockItemConfiguration->isEnableQtyIncrements()
+                && $this->mathDivision->getExactDivision($requestedQty, $stockItemConfiguration->getQtyIncrements()))) {
```

Comment:
this check looks excessive. 
As StockItemConfiguration which proxy call to legacy StockItemInterface has a method 
`  public function getQtyIncrements(): float;`

that's why initial check currently looks like 

```
    {
        $result = new \Magento\Framework\DataObject();
        if ($stockItem->getSuppressCheckQtyIncrements()) {
            return $result;
        }

        $qtyIncrements = $stockItem->getQtyIncrements() * 1;

        if ($qtyIncrements && $this->mathDivision->getExactDivision($qty, $qtyIncrements) != 0) {
            $result->setHasError(true)
                ->setQuoteMessage(__('Please correct the quantity for some products.'))
                ->setErrorCode('qty_increments')
                ->setQuoteMessageIndex('qty');
            if ($stockItem->getIsChildItem()) {
                $result->setMessage(
                    __(
                        'You can buy %1 only in quantities of %2 at a time.',
                        $stockItem->getProductName(),
                        $qtyIncrements
                    )
                );
            } else {
                $result->setMessage(__('You can buy this product only in quantities of %1 at a time.', $qtyIncrements));
            }
        }
        return $result;
```

So, taking $stockItemConfiguration->getQtyIncrements() should be enough 


This function under the hood looks like 
```php
    public function getQtyIncrements()
    {
        if ($this->qtyIncrements === null) {
            if ($this->getEnableQtyIncrements()) {
                if ($this->getUseConfigQtyIncrements()) {
                    $this->qtyIncrements = $this->stockConfiguration->getQtyIncrements($this->getStoreId());
                } else {
                    $this->qtyIncrements = (int) $this->getData(static::QTY_INCREMENTS);
                }
            }
            if ($this->qtyIncrements <= 0) {
                $this->qtyIncrements = false;
            }
        }
        return $this->qtyIncrements;
    }
```

Reply:
I would change it to the following (note: I added the explicit strict comparison with `0`)
```
if ($this->mathDivision->getExactDivision($requestedQty, $this->configuration->getQtyIncrements()) !== 0
            || $this->mathDivision->getExactDivision(
                $requestedQty,
                $stockItemConfiguration->getQtyIncrements()
            ) !== 0) {
            return false;
        }
```